Packs/1Password/Integrations/OnePasswordEventCollector/OnePasswordEventCollector.py

```diff
@@ -0,0 +1,392 @@
+                continue
+
+            if len(events) == max_events:
+                break
+
+            add_fields_to_event(event, event_type)
+            events.append(event)
+            ids_to_skip.add(event_id)
+
+        demisto.debug(
+            f'Response has {len(response["items"])} events and pagination cursor: {pagination_cursor}.'
+            f'Used client arguments: {client_kwargs}.'
+        )
+
+        # Followup API calls need to include pagination cursor (if any more events)
+        has_more_events = False if len(events) == max_events else response['has_more']
+        client_kwargs = {'event_type': event_type, 'pagination_cursor': pagination_cursor}
+
+    return events
+
+
+def push_events(events: list[dict], next_run: dict[str, Any] | None = None) -> None:
+    """Sends events to XSIAM and optionally sets next run (if `next_run` is given).
+
+    Args:
+        events (list): List of event dictionaries.
+        next_run (dict | None): Optional next run dictionary of all event types. Defaults to None.
+    """
+    demisto.debug(f'Sending {len(events)} to XSIAM')
+    send_events_to_xsiam(events=events, vendor=VENDOR, product=PRODUCT)
+
+    if next_run:
+        demisto.debug(f'Setting next run to {next_run}')
+        demisto.setLastRun(next_run)
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def fetch_events(
+    client: Client,
+    event_type: str,
+    first_fetch_date: datetime,
+    type_last_run: dict[str, Any],
+    max_events: int
+) -> tuple[dict, list]:
+    """Fetches new events via 1Password Events API client based on a specified event type.
+
+    Args:
+        client (Client): 1Password Events API client.
+        event_type (str): Type of 1Password event (e.g. 'Item usage actions', 'Sign in attempts').
+        first_fetch_date (datetime): Datetime from which to get events.
+        type_last_run (dict): Dictionary of the last run (if any) of the event type with optional 'from_date' and 'ids' list.
+        max_events (int): The maximum number of events to fetch.
+
+    Returns:
+        tuple[dict, list]: Dictionary of the next run of the event type with 'from_date' and 'ids' list, list of fetched events.
+    """
+    last_run_ids = set(type_last_run.get('ids') or [])
+    from_date = arg_to_datetime(type_last_run.get('from_date')) or first_fetch_date
+
+    demisto.debug(f'Fetching events of type: {event_type} from date: {from_date.isoformat()}')
+
+    events = get_events_from_client(
+        client=client,
+        event_type=event_type,
+        from_date=from_date,
+        max_events=max_events,
+        ids_to_skip=last_run_ids,
+    )
+
+    # API returns events sorted by timestamp in ascending order (oldest to newest), so last event has max timestamp
+    max_timestamp = events[-1]['timestamp'] if events else None
+    next_run_ids = [event['uuid'] for event in events if event['timestamp'] == max_timestamp]
+    type_next_run = {'from_date': max_timestamp, 'ids': next_run_ids}
```

Comment:
`type_next_run`? What type?

Reply:
Updated variable names to be more understandable in commit bcef245a24f5d8f7ad476073a3c717f090e72bba